fix: Setting Parse Server option `masterKeyIps: []` doesn't disable all IP addresses

stewones opened the pull request:
### New Pull Request Checklist
<!--
    Please check the following boxes [x] before submitting your issue.
    Click the "Preview" tab for better readability.
    Thanks for contributing to Parse Server!
-->

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [ ] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

it's not quite working as expected, default ips are being concatenated to the user option. this PR aims to fix it.

### Approach
<!-- Add a description of the approach in this PR. -->

### TODOs before merging
<!--
    Add TODOs that need to be completed before merging this PR.
    Delete suggested TODOs that do not apply to this PR.
-->

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK <!-- no hard-coded error codes in Parse Server -->

stewones committed:
fix: add ability to disable masterKeyIps

stewones committed:
Merge branch 'alpha' into fix/masterKeyIps

dblythy commented on src/ParseServer.js:
```diff
@@ -468,9 +468,12 @@ function injectDefaults(options: ParseServerOptions) {
-  options.masterKeyIps = Array.from(
-    new Set(options.masterKeyIps.concat(defaults.masterKeyIps, options.masterKeyIps))
-  );
+  options.masterKeyIps =
+    options.masterKeyIps?.length > 0
+      ? Array.from(
+        new Set(options.masterKeyIps.concat(defaults.masterKeyIps, options.masterKeyIps))
+      )
+      : [];
```
```suggestion
```

I think these lines can be removed completely

stewones replied:
yeah looking closer, fully agreed 😄

I was actually confused initially and didn't want to make some shit, especially because I'm not being able to run tests locally due to this error. any ideas?

```
~/dev/parse-server fix/masterKeyIps
❯ npm test

> parse-server@6.0.0-alpha.12 pretest /Users/stewan/dev/parse-server
> cross-env MONGODB_VERSION=${MONGODB_VERSION:=5.3.2} MONGODB_TOPOLOGY=${MONGODB_TOPOLOGY:=standalone} mongodb-runner start

  ◝ Starting a MongoDB deployment to test against...Error: Could not find download URL for version 5.3.2 {
  version: '5.3.2',
  arch: [ 'arm64', 'aarch64' ],
  platform: 'darwin',
  target: [
    { value: 'osx', priority: 1 },
    { value: 'osx-ssl', priority: 10 },
    { value: 'darwin', priority: 1 },
    { value: 'macos', priority: 1 }
  ],
  enterprise: false,
  cryptd: false
}
    at resolve (/Users/stewan/dev/parse-server/node_modules/mongodb-download-url/lib/index.js:177:15)
    at async getDownloadURL (/Users/stewan/dev/parse-server/node_modules/mongodb-download-url/lib/index.js:242:12)
npm ERR! Test failed.  See above for more details.
```

dblythy commented on src/middlewares.js:
```diff
@@ -166,7 +166,11 @@ export function handleParseHeaders(req, res, next) {
 
   let isMaster = info.masterKey === req.config.masterKey;
-  if (isMaster && !ipRangeCheck(clientIp, req.config.masterKeyIps || [])) {
+  if (
+    isMaster &&
+    req.config.masterKeyIps?.length &&
+    !ipRangeCheck(clientIp, req.config.masterKeyIps || [])
+  ) {
```
```suggestion
if (isMaster && !ipRangeCheck(clientIp, req.config.masterKeyIps || [])) {
```

dblythy reviewed (changes requested):
I think the logic here is:

`masterKeyIps: []`: disable masterKey from anywhere
`masterKeyIps: ["0.0.0.0/0"]`: allow masterKey from anywhere

stewones committed:
Update src/ParseServer.js

masterKeyIps: []: disable masterKey from anywhere
masterKeyIps: ["0.0.0.0/0"]: allow masterKey from anywhere

Co-authored-by: Daniel <daniel-blyth@live.com.au>

stewones committed:
Update src/middlewares.js

Co-authored-by: Daniel <daniel-blyth@live.com.au>

mtrezza commented:
I'm closing this PR since the way to "disable" or allow all IPs is to set `masterKeyIps: ['0.0.0.0/0']`. That is also documented in the option:

> - Setting `['0.0.0.0/0']` means disabling the filter 

The reason you probably haven't seen this in the option docs is that the docs are only created for stable releases, so once Parse Sever 6.0.0 is released, the docs will also be updated.

mtrezza commented on src/middlewares.js:
```diff
@@ -166,7 +166,7 @@ export function handleParseHeaders(req, res, next) {
 
   let isMaster = info.masterKey === req.config.masterKey;
-  if (isMaster && !ipRangeCheck(clientIp, req.config.masterKeyIps || [])) {
+if (isMaster && !ipRangeCheck(clientIp, req.config.masterKeyIps || [])) {
```
Could you run `npm run prettify && npm run lint-fix`?

stewones committed:
💅

mtrezza commented on src/ParseServer.js:
```diff
@@ -467,10 +467,6 @@ function injectDefaults(options: ParseServerOptions) {
     }
   });
-
-  options.masterKeyIps = Array.from(
```
Can this really just be removed and the issue is fixed?

cc @dblythy

dblythy replied:
I believe so. I will test and report back
